### PR TITLE
Chore: Keeping the list of plugin executable complete

### LINKED ISSUE #52640: Grafana: make all backend plugin platform files executabe
**Why?**
Currently we only mark some architecture types of backend plugins as executable. https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/installer/installer.go#L633-L635

**Desired state**
We should mark all the backend plugin architecture types as executable - e.g. `_linux_arm64` and `_darwin_arm64`. Also check if `_windows_amd64` also needs to be marked as executable. 

**Additionally**
If possible add the permissions check to [plugin-validator](https://github.com/grafana/plugin-validator) `plugincheck2` and, if necessary, adjust the [report parsing in the pipeline](https://github.com/grafana/community-pipeline/blob/main/pkg/lib/report.go#L148)

**Backport**
- Yes, latest `8.x` and `9.x` version

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**

Extending the list of backend plugin files we `chmod` to `755` to make it complete

**Why do we need this feature?**

Might not be needed in the future if we manage to validate which plugins were not packaged properly and fix them since we do not accept new submissions with incorrect permissions.

**Who is this feature for?**
Patching for improperly packaged plugins

**Which issue(s) does this PR fix?**:

Fixes #52640

**Special notes for your reviewer**:

### CODE REVIEW

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

File: pkg/plugins/storage/fs.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func isSymlinkRelativeTo(basePath string, symlinkDestPath string, symlinkOrigPat
 func extractFile(file *zip.File, filePath string) (err error) {
 	fileMode := file.Mode()
 	// This is entry point for backend plugins so we want to make them executable
-	if strings.HasSuffix(filePath, "_linux_amd64") || strings.HasSuffix(filePath, "_darwin_amd64") {
+	if strings.HasSuffix(filePath, "_linux_amd64") || strings.HasSuffix(filePath, "_linux_arm") || strings.HasSuffix(filePath, "_linux_arm64") || strings.HasSuffix(filePath, "_darwin_amd64") || strings.HasSuffix(filePath, "_darwin_arm64") || strings.HasSuffix(filePath, "_windows_amd64") {
```
[the remark this answers is not in the document]
Comment (wbrowne) [reply pkg/plugins/storage/fs.go:223]: Well that was a fun adventure. Looks like it does what we need, with a small change needed!

**Review by wbrowne — COMMENTED** [review]
Looks like this fails [some tests](https://github.com/grafana/grafana/blob/main/pkg/plugins/storage/fs_test.go#L139) now that were protecting this behaviour from changing. Looping in @marefr to verify what makes sense to do

**Comment by marefr** [comment]
```
--- FAIL: TestExtractFiles (0.00s)
    --- FAIL: TestExtractFiles/Should_preserve_file_permissions_for_plugin_backend_binaries_for_linux_and_darwin (0.00s)
        fs_test.go:160: 
            	Error Trace:	/drone/src/pkg/plugins/storage/fs_test.go:160
            	Error:      	Not equal: 
            	            	expected: "-rw-r--r--"
            	            	actual  : "-rwxr-xr-x"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	--rw-r--r--
            	            	+-rwxr-xr-x
            	Test:       	TestExtractFiles/Should_preserve_file_permissions_for_plugin_backend_binaries_for_linux_and_darwin
FAIL
```



> Looks like this fails [some tests](https://github.com/grafana/grafana/blob/main/pkg/plugins/storage/fs_test.go#L139) now that were protecting this behaviour from changing. Looping in @marefr to verify what makes sense to do

Yep need to adjust the test for Windows

**Comment by tolzhabayev** [comment]
Interesting, why would it expect `-rw-r--r--` in windows? @marefr @wbrowne

**Comment by marefr** [comment]
> Interesting, why would it expect `-rw-r--r--` in windows? @marefr @wbrowne

Because the code before these changes only set 0755 to linux_amd64 and darwin_amd64 and with your changes 0755 applies to "all" executables

**Comment by tolzhabayev** [comment]
> > Interesting, why would it expect `-rw-r--r--` in windows? @marefr @wbrowne
> 
> Because the code before these changes only set 0755 to linux_amd64 and darwin_amd64 and with your changes 0755 applies to "all" executables

I get that part, I am wondering if maybe there is a good reason why it should not be `0755`?

**Comment by marefr** [comment]
> I get that part, I am wondering if maybe there is a good reason why it should not be `0755`?

I guess we need to test it on Windows to figure out

**Comment by wbrowne** [comment]
That's what I tested and it seemed fine to me!

**Comment by marefr** [comment]
> That's what I tested and it seemed fine to me!

:boom:

**Pull request merged**